### DISCUSSION TITLE
Provide `onMutate` on `MutationCache`

### DISCUSSION THREAD

**Discussion by stefanprobst** [TanStack/query#3035] [Ideas]
hi, what do you think about a global `onMutate` cb on the `MutationCache` (in addition to `onSuccess` and `onError`)? i think it could be helpful for e.g. managing toast notifications for mutations in one place.

**Comment by TkDodo** [comment]
hm, is it really a use-case to show a toast notification when a mutation starts? I mean maybe, since react-hot-toasts allows for loading spinners inside the toast 🤔

**Reply by stefanprobst** [reply]
yes i was thinking about react-hot-toasts's `toast.promise`

**Comment by TkDodo** [comment]
I think I'd accept a PR in this direction :)
